android-core/src/androidTest/kotlin/com.mparticle/networking/MParticleBaseClientImplTest.kt

```diff
@@ -20,11 +21,150 @@ class MParticleBaseClientImplTest : BaseCleanInstallEachTest() {
+        val configUrl = mRandomUtils.getAlphaString(20)
+        val audienceUrl = mRandomUtils.getAlphaString(20)
+        val eventsUrl = mRandomUtils.getAlphaString(20)
+        val options = MParticleOptions.builder(mContext)
+            .credentials(apiKey, "secret")
+            .networkOptions(
+                NetworkOptions.builder()
+                    .addDomainMapping(
+                        DomainMapping.audienceMapping(audienceUrl)
+                            .build()
+                    )
+                    .addDomainMapping(
+                        DomainMapping.configMapping(configUrl)
+                            .build()
+                    )
+                    .addDomainMapping(
+                        DomainMapping.identityMapping(identityUrl)
+                            .build()
+                    )
+                    .addDomainMapping(DomainMapping.eventsMapping(eventsUrl).build())
+                    .build()
+            )
+            .build()
+        MParticle.start(options)
+        val baseClientImpl = AccessUtils.getApiClient() as MParticleBaseClientImpl
+        val url =
+            baseClientImpl.getPodUrl(NetworkOptionsManager.MP_URL_PREFIX, prefix, true)
+        assertEquals("${NetworkOptionsManager.MP_URL_PREFIX}.$prefix.mparticle.com", url)
+    }
+
+    @Test
+    fun testUrlPrefixWithoutPodRedirection() {
+        val prefix = "eu1"
+        val identityUrl = mRandomUtils.getAlphaString(20)
+        val configUrl = mRandomUtils.getAlphaString(20)
+        val audienceUrl = mRandomUtils.getAlphaString(20)
+        val eventsUrl = mRandomUtils.getAlphaString(20)
+        val options = MParticleOptions.builder(mContext)
+            .credentials(apiKey, "secret")
+            .networkOptions(
+                NetworkOptions.builder()
+                    .addDomainMapping(
+                        DomainMapping.audienceMapping(audienceUrl)
+                            .build()
+                    )
+                    .addDomainMapping(
+                        DomainMapping.configMapping(configUrl)
+                            .build()
+                    )
+                    .addDomainMapping(
+                        DomainMapping.identityMapping(identityUrl)
+                            .build()
+                    )
+                    .addDomainMapping(DomainMapping.eventsMapping(eventsUrl).build())
+                    .build()
+            )
+            .build()
+        MParticle.start(options)
+        val baseClientImpl = AccessUtils.getApiClient() as MParticleBaseClientImpl
+        val url =
+            baseClientImpl.getPodUrl(NetworkOptionsManager.MP_URL_PREFIX, prefix, false)
+        assertEquals("${NetworkOptionsManager.MP_URL_PREFIX}.mparticle.com", url)
+    }
+
+    @Test
+    fun testAllPrefixes() {
+        val map = mapOf<String, String>(
+            Pair("us1-1vc4gbp24cdtx6e31s58icnymzy83f1uf", "us1"),
```

Comment:
just so future devs know, add a comment above here that says these are fake api keys.

i would also test one where there is no prefix, so that legacy us1 is covered